data/transactions/logic/assembler.go

```diff
@@ -1612,6 +1645,57 @@ func disBytecblock(dis *disassembleState, spec *OpSpec) {
 	_, dis.err = dis.out.Write([]byte("\n"))
 }
 
+func allPrintableAscii(bytes []byte) bool {
+	for _, b := range bytes {
+		if b < 32 || b > 126 {
+			return false
+		}
+	}
+	return true
+}
+func guessByteFormat(bytes []byte) string {
+	var short basics.Address
+
+	if len(bytes) == len(short) {
+		copy(short[:], bytes[:])
+		return fmt.Sprintf("addr %s", short.String())
+	}
+	if allPrintableAscii(bytes) {
+		return fmt.Sprintf("\"%s\"", string(bytes))
+	}
+	return "0x" + hex.EncodeToString(bytes)
+}
+
+func disBytec(dis *disassembleState, spec *OpSpec) {
+	lastIdx := dis.pc + spec.Details.Size - 1
+	if len(dis.program) <= lastIdx {
+		missing := lastIdx - len(dis.program) + 1
+		dis.err = fmt.Errorf("unexpected %s opcode end: missing %d bytes", spec.Name, missing)
+		return
+	}
+	dis.nextpc = dis.pc + spec.Details.Size
+	var suffix string
+	var b uint
+	switch spec.Opcode {
+	case 0x28:
+		suffix = "_0"
+		b = 0
+	case 0x29:
+		suffix = "_1"
+		b = 1
+	case 0x2a:
+		suffix = "_2"
+		b = 2
+	case 0x2b:
+		suffix = "_3"
+		b = 3
+	case 0x27:
+		b = uint(dis.program[dis.pc+1])
+		suffix = fmt.Sprintf(" %d", b)
+	}
+	_, dis.err = fmt.Fprintf(dis.out, "bytec%s // %s\n", suffix, guessByteFormat(dis.byteConstants[b]))
```

Comment:
I like this structure better, with one printf, can we do this for disIntc too?

Reply:
Yup, done now, with a refactor to get better dryrun output too.